cub/iterator/tex_ref_input_iterator.cuh

```diff
@@ -42,6 +42,18 @@
 #include "../util_debug.cuh"
 #include "../config.cuh"
 
+// This class needs to go through a deprecation cycle and be removed, as the
+// underlying cudaBindTexture / cudaUnbindTexture APIs are deprecated.
+// See issue NVIDIA/cub#191.
+// Turn off deprecation warnings when compiling this file until then.
+#if CUB_HOST_COMPILER == CUB_HOST_COMPILER_MSVC
+#pragma warning(disable:4996)
+#elif CUB_HOST_COMPILER == CUB_HOST_COMPILER_GCC || \
+      CUB_HOST_COMPILER == CUB_HOST_COMPILER_CLANG
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+
```

Comment:
I would deprecate or remove this entirely.

Reply:
#191